src/main/java/org/klodnicki/service/BookService.java

```diff
@@ -73,6 +73,15 @@ private List<BookInfo> getAllBooksFromDatabaseSortByParameter(String parameter)
         return bookRepository.findAllBooksSortByParameter(parameter);
     }
 
+    private String prepareParameterToDatabase(String parameter) {
```

Comment:
This method shouldn't return "parameter" if none are found, right? I think this should be changed unless I don't see something. Let me know!

Reply:
it is stopped later by the validation method sortOptionValidation(String parameter)
![image](https://github.com/BartoliniAndBorderCollies/Library_Manager_Official/assets/126821059/4c40352b-14c8-450f-bc0d-ae12cb196683)

and then by using this method in here:

![image](https://github.com/BartoliniAndBorderCollies/Library_Manager_Official/assets/126821059/08e6307c-b0a4-492a-9e75-92bd5d95fec7)
-----------------------------------------------------------------------------------------------------------------------------------
but of course this validation can be done in prepareParameterToDatabase(String parameter)

let me know what is better. stays like it is or shall I change prepareParameterToDatabase()?

Reply:
So there are two methods that should be used together. This is probably a case to create a separate service class for handling these parameters which would make this super clear then and separate functions (logic for books and logic for parameters) would be decoupled.

This would look much better if validating parameter and then returning hqlParameter were in one method (that means use one method which will call another method for first validating and then returning parameter).

Reply:
I made a change that we were talking about - one line of code instead of whole method